### PR TITLE
update configuration for rootless podman

### LINKED ISSUE #6777: rootless config files are not created
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

The config files ~/.config/containers/storage.conf and ~/.config/containers/libpod.conf do not exist. But podman info --debug refers to them. But the [rootless tutorial](https://github.com/containers/libpod/blob/master/docs/tutorials/rootless_tutorial.md#user-configuration-files) indicates that they should exist. Is there some step, that I miss, to create these files? Or is the tutorial outdated? From [this discussion](https://lists.podman.io/archives/list/podman@lists.podman.io/thread/Z4IMPJCZRYXIGQUSELRBUM6L7533J5IG/) I get, that `podman info` should create the config files?

My $XDG_CONFIG_HOME is not set by default on my server. So ${XDG_CONFIG_HOME}/containers becomes /containers, where normal users do not have any rights. But I tested with a new user, where I added `export XDG_CONFIG_HOME=/home/netter2/.config` to the .bashrc before the first login. `podman info` did not create anything. I also tried to create the folder .config/containers beforehand, which did not work either.

Is there maybe some global config, that forbids rootless to create their own config files?

**Steps to reproduce the issue:**

1. Create a new user and login with that user

2. run `podman info`

3. check whether .config/containers/storage.conf exists or does not exist

**Describe the results you received:**
~/.config/containers/storage.conf and ~/.config/containers/libpod.conf do not exist

**Describe the results you expected:**
~/.config/containers/storage.conf and ~/.config/containers/libpod.conf should exist and be populated with a standard configuration.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
Version:      2.0.0
API Version:  1
Go Version:   go1.13.4
Built:        Thu Jan  1 01:00:00 1970
OS/Arch:      linux/amd64

```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.15.0
  cgroupVersion: v1
  conmon:
    package: conmon-2.0.18-1.el8.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.18, commit: 3f215ac67ed650ab6a44f8c036bcbd692afdc21e'
  cpus: 32
  distribution:
    distribution: '"centos"'
    version: "8"
  eventLogger: file
  hostname: ****
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1039
      size: 1
    - container_id: 1
      host_id: 951968
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1037
      size: 1
    - container_id: 1
      host_id: 951968
      size: 65536
  kernel: 4.18.0-147.8.1.el8_1.x86_64
  linkmode: dynamic
  memFree: 20184137728
  memTotal: 270112145408
  ociRuntime:
    name: runc
    package: runc-1.0.0-15.5.el8.x86_64
    path: /usr/bin/runc
    version: |-
      runc version 1.0.0-rc10
      commit: ffb456f24bd637cec53c611f309d89abb027b1d3
      spec: 1.0.1-dev
  os: linux
  remoteSocket:
    path: /run/user/1037/podman/podman.sock
  rootless: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.1.1-1.el8.x86_64
    version: |-
      slirp4netns version 1.1.1
      commit: bbf27c5acd4356edb97fa639b4e15e0cd56a39d5
      libslirp: 4.2.0
      SLIRP_CONFIG_VERSION_MAX: 2
  swapFree: 3010519040
  swapTotal: 4294963200
  uptime: 410h 28m 20.54s (Approximately 17.08 days)
registries: {}
store:
  configFile: /home/myuser/.config/containers/storage.conf
  containerStore:
    number: 2
    paused: 0
    running: 0
    stopped: 2
  graphDriverName: overlay
  graphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: fuse-overlayfs-1.1.1-1.el8.x86_64
      Version: |-
        fusermount3 version: 3.2.1
        fuse-overlayfs: version 1.1.0
        FUSE library version 3.2.1
        using FUSE kernel interface version 7.26
  graphRoot: /home/myuser/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 1
  runRoot: /run/user/1037/containers
  volumePath: /home/myuser/.local/share/containers/storage/volumes
version:
  APIVersion: 1
  Built: 0
  BuiltTime: Thu Jan  1 01:00:00 1970
  GitCommit: ""
  GoVersion: go1.13.4
  OsArch: linux/amd64
  Version: 2.0.0

```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
Installierte Pakete
podman.x86_64        2.0.0-2.el8                                                                  @devel_kubic_libcontainers_stable
Verfügbare Pakete
podman.aarch64        2.0.0-2.el8                                                                  devel_kubic_libcontainers_stable
podman.src        2.0.0-2.el8                                                                  devel_kubic_libcontainers_stable

```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
CentOS 8 on a physical workstation.

### PR DESCRIPTION
**Pull request by gatoniel**
I updated the configuration part of the tutorial on rootless podman. I added the order in which configuration files are read in and a hint, how users can create default configuration in the home directories.

Closes #6777

Signed-off-by: Niklas Netter <niknett@gmail.com>

### CODE REVIEW

File: docs/tutorials/rootless_tutorial.md [hunk 1]
```diff
@@ -106,8 +106,31 @@ Once the Administrator has completed the setup on the machine and then the confi
 
 ### User Configuration Files
 
-The Podman configuration files for root reside in `/usr/share/containers` with overrides in `/etc/containers`.  In the rootless environment they reside in `${XDG_CONFIG_HOME}/containers` (usually `~/.config/containers`) and are owned by each individual user.  The main files are `libpod.conf` and `storage.conf` and the user can modify these files as they wish.
+The three main configuration files are `containers.conf`, `storage.conf` and `registries.conf`.
+#### containers.conf
```
Comment (mheon) [inline docs/tutorials/rootless_tutorial.md:110]: Can you add an extra blank line before this header?

File: docs/tutorials/rootless_tutorial.md [hunk 2]
```diff
@@ -106,8 +106,31 @@ Once the Administrator has completed the setup on the machine and then the confi
...
+
+In rootless podman certain fields in `/etc/containers/storage.conf` are ignored.
+
+#### registries
+Registry configuration is read in by this order
+1. `/etc/containers/registries.conf`
+2. `/etc/containers/registries.d/*`
+3. `HOME/.config/containers/registries.conf`
+
+The files in the home directory should be used to configure rootless podman for personal needs. These files do not exist by default. Users can copy the files from `/usr/share/containers` or `/etc/containers` and modify them.
```
Comment (mheon) [inline docs/tutorials/rootless_tutorial.md:131]: Suggest rephrasing: "do not exist by default" -> "are not created by default"

**Comment by mheon** [comment]
Thanks for the contribution! Your commit appears to be missing a signoff - can you run `git commit -s --amend --no-edit` to add one and re-push?

**Comment by gatoniel** [comment]
Sorry for the mess. There should now be a signoff.

File: docs/tutorials/rootless_tutorial.md [hunk 3]
```diff
@@ -106,8 +106,32 @@ Once the Administrator has completed the setup on the machine and then the confi
...
+
+In rootless podman certain fields in `/etc/containers/storage.conf` are ignored.
+
+#### registries
+Registry configuration is read in by this order
+1. `/etc/containers/registries.conf`
+2. `/etc/containers/registries.d/*`
+3. `HOME/.config/containers/registries.conf`
+
+The files in the home directory should be used to configure rootless podman for personal needs. These files are not created by default. Users can copy the files from `/usr/share/containers` or `/etc/containers` and modify them.
```
Comment (TomSweeneyRedHat) [inline docs/tutorials/rootless_tutorial.md:132]: I thought some of these files were created by default for a rootless user in their home directory?

Comment (rhatdan) [reply docs/tutorials/rootless_tutorial.md:132]: Not any longer.

**Comment by rhatdan** [comment]
@gatoniel You need to rebase and squash your commits as well as sign them

git rebase -i origin
git commit -a --amend -s
git push --force

File: docs/tutorials/rootless_tutorial.md [hunk 4]
```diff
@@ -106,8 +106,32 @@ Once the Administrator has completed the setup on the machine and then the confi
...
+3. `$HOME/.config/containers/containers.conf`
+
+if they exist in that order. Each file can override the previous for particular fields.
+
+#### storage.conf
+For `storage.conf` the order is
+1. `/etc/containers/storage.conf`
+2. `$HOME/.config/containers/storage.conf`
+
+In rootless podman certain fields in `/etc/containers/storage.conf` are ignored.
```
Comment (rhatdan) [inline docs/tutorials/rootless_tutorial.md:124]: We should list the fields that are ignored.
```
       graphroot=""
         container storage graph dir (default: "/var/lib/containers/storage")
         Default directory to store all writable content created by container storage programs.

       runroot=""
         container storage run dir (default: "/var/run/containers/storage")
         Default directory to store all temporary writable content created by container storage programs.
```

File: docs/tutorials/rootless_tutorial.md (hunk 3 above)
Comment (rhatdan) [reply docs/tutorials/rootless_tutorial.md:132]: Perhaps we should create the directories.  The problem with creating the files, is we ended up hard coding some defaults that we later changed, and users ended up with bad files in their homedirs, which we could not change.

**Comment by rhatdan** [comment]
BTW @gatoniel Nice work.

File: docs/tutorials/rootless_tutorial.md (hunk 4 above)
Comment (gatoniel) [reply docs/tutorials/rootless_tutorial.md]: I suggest adding their default values, when in rootless podman. (see new commits). I do not exactly know, if podman really reads out $XDG_RUNTIME_DIR or just defaults to `/run/user/$UID`.

Comment (rhatdan) [reply docs/tutorials/rootless_tutorial.md]: It uses XDG_RUNTIME_DIR

**Comment by rhatdan** [comment]
You really need to squash your commits and sign the one commit, to get it past the gating test.

**Comment by rhatdan** [comment]
@gatoniel still working on this?  Needs a rebase and to squash your commits.

**Comment by gatoniel** [comment]
> 
> 
> @gatoniel still working on this? Needs a rebase and to squash your commits.

I am so sorry for taking your time. I have never used the rebase before. Now I swapped pick -> squash . I hope that´s correct.

How can I resolve the build_each_commit fail?

**Comment by rhatdan** [comment]
Sadly still needs a rebase.

**Comment by rhatdan** [comment]
@gatoniel Sadly one more rebase.

**Comment by gatoniel** [comment]
I do not think this is a rebase issue. There is always an auto-merging conflict.

I now resolved the auto-merging conflict by using Githubs PR resolver, this added the new commit. The tide test is now pending instead of failing. But build_each_commit is still failing with this error message

> source $SCRIPT_BASE/lib.sh &> /dev/null
> git fetch --depth 50 origin $DEST_BRANCH |& ${TIMESTAMP}
> [11:38:38] START - All [+xxxx] lines that follow are relative to right now.
> [+0003s] From https://github.com/containers/podman
> [+0003s]  * branch                master     -> FETCH_HEAD
> [+0003s]  * [new branch]          master     -> origin/master
> [11:38:41] END - [+0003s] total duration since START
> make build-all-new-commits GIT_BASE_BRANCH=origin/$DEST_BRANCH |& ${TIMESTAMP}
> [11:38:41] START - All [+xxxx] lines that follow are relative to right now.
> [+0004s] # Validate that all the commits build on top of origin/master
> [+0004s] git rebase origin/master -x make
> [+0006s] Rebasing (1/2)
> error: could not apply c35c453bf... update configuration for rootless podman
> [+0006s] Resolve all conflicts manually, mark them as resolved with
> [+0006s] "git add/rm <conflicted_files>", then run "git rebase --continue".
> [+0006s] You can instead skip this commit: run "git rebase --skip".
> [+0006s] To abort and get back to the state before "git rebase", run "git rebase --abort".
> [+0006s] Could not apply c35c453bf... update configuration for rootless podman
> [+0006s] Auto-merging docs/tutorials/rootless_tutorial.md
> [+0006s] CONFLICT (content): Merge conflict in docs/tutorials/rootless_tutorial.md
> [+0006s] Adding contrib/systemd/user
> [+0006s] make: *** [Makefile:695: build-all-new-commits] Error 1
> [11:38:47] END - [+0006s] total duration since START
> 
> Exit status: 2

I really do not know what to do.

**Comment by rhatdan** [comment]
@gatoniel I took this over to complete it #7079
Thanks, I will get it merged.

**Pull request closed without merging**